### DISCUSSION TITLE
How do I create name-value pairs for SingleChoice elements?

### DISCUSSION THREAD

**Discussion by storyteller-dev** [ctreffe/alfred#143] [Alfred Q&A] [answered]
I'd like to make the value of a select field different from its label. So for example instead of:

```Html
<select class="form-control input-field SingleChoiceList-element" id="gender" style=" " name="gender">
     <option value="" id="gender_choice1" name="gender"></option>
     <option value="Männlich" id="gender_choice2" name="gender">Männlich</option>
     <option value="Weiblich" id="gender_choice3" name="gender">Weiblich</option>
     <option value="Divers" id="gender_choice4" name="gender">Divers</option>
</select>
```

I'd like to render the select field with different values than the labels:

```Html
<select class="form-control input-field SingleChoiceList-element" id="gender" style=" " name="gender">
     <option value="" id="gender_choice1" name="gender"></option>
     <option value="male" id="gender_choice2" name="gender">Männlich</option>
     <option value="female" id="gender_choice3" name="gender">Weiblich</option>
     <option value="Divers" id="gender_choice4" name="gender">Divers</option>
</select>
```

How can I achieve that using a SingleChoice element?

**Comment by jobrachem** [comment, marked as answer]
Alfred3 currently does not provide that possibility. 

Alfred3 will internally translate the values of most `SingleChoice` type elements to integers, counting up from 1. There's one exception, namely `SingleChoiceList`, which does not translate the value to an integer but uses the label string directly.

Maybe it would be useful to provide the SingleChoice type elements with an additional parameter that allows the specification of the value-label-association. While this does add complexity, it may still pay off notably by improving the handling of choice data both within an experiment and in the resulting data set.

**Reply by storyteller-dev** [reply]
I thought as much and while I tried around with this, two possible annotations crossed my mind when trying out how to do it.

The one being passing a dictionary:

```Python
import alfred3 as al

exp = al.Experiment()


exp += al.Page(name="demographics")
exp.demographics += al.SingleChoiceList(
    {
        "": "",
        "male": "Männlich",
        "female": "Weiblich",
        "non-binary": "Divers",
    }
    leftlab="Geschlecht",
    name="gender",
    force_input=True,
)
```

The second was a Text element with a name:

```Python
import alfred3 as al

exp = al.Experiment()


exp += al.Page(name="demographics")
exp.demographics += al.SingleChoiceList(
    al.Text("", name=""),
    al.Text("Männlich", name="male"),
    al.Text("Weiblich", name="female"),
    al.Text("Divers", name="non-binary"),
    leftlab="Geschlecht",
    name="gender",
    force_input=True,
)
```

While the first option is - at least for me - the most intuitive, the second would nicely integrate in the existing Alfred3 documentation that describes the possibility of passing elements as choice labels.
